cmake/test.cmake

```diff
@@ -79,3 +79,33 @@ function(add_unit_test NAME)
   add_cpp_test(${NAME} SRCS ${ARG_SRCS} LIBS ${ARG_LIBS} DEPS ${ARG_DEPS})
   add_dependencies(unit_tests "${NAME}_test")
 endfunction(add_unit_test)
+
+# For tests that use eBPF components and need to run in a container that supports eBPF. Does everything in add_unit_test()
+# and labels test with the eBPFContainer label.  These tests will not be run by default, but only if the RUN_EBPF_TESTS
+# environment variable is set to TRUE.
+function(add_ebpf_unit_test NAME)
+  set(TEST_NAME "${NAME}_test")
+  cmake_parse_arguments(ARG "" "" "SRCS;LIBS;DEPS" ${ARGN})
+
+  add_executable(
+    ${TEST_NAME}
+      "${TEST_NAME}.cc"
+      ${ARG_SRCS}
+  )
+
+  # https://stackoverflow.com/questions/61572034/ctest-disable-a-set-of-labeled-tests-by-default-but-run-them-when-explicitly
+  add_test(NAME ${TEST_NAME} COMMAND sh -c "if [ \"\${RUN_EBPF_TESTS:-}\" ]; then $<TARGET_FILE:${TEST_NAME}>; else exit 127; fi")
+  set_tests_properties(${TEST_NAME} PROPERTIES SKIP_RETURN_CODE 127)
```

Comment:
looks like you also implemented test LABELS below (like in https://stackoverflow.com/questions/24495412/ctest-using-labels-for-different-tests-ctesttestfile-cmake)

Do we still need the env-variable based selection? (if not, can simplify and remove)

Also if it's not required, we might reduce duplicated code with `add_cpp_test` by
1. Adding a LABELS argument to `cmake_parse_arguments` in `add_cpp_test`
2. `add_ebpf_unit_test` will then be syntactic sugar of calling `add_cpp_test` with the `eBPFContainer` label

A little copy paste won't kill us here, so not mandatory.. (just thinking we could prevent a future introduction of a discrepancy between tests on future modification of `add_{ebpf_unit,cpp}_test`)